libafl_qemu/src/modules/calls.rs

```diff
@@ -543,19 +543,19 @@ static mut CALLSTACKS: Option<ThreadLocal<UnsafeCell<Vec<GuestAddr>>>> = None;
 #[derive(Debug)]
 pub struct FullBacktraceCollector {}
 
-impl Default for FullBacktraceCollector {
-    fn default() -> Self {
-        Self::new()
-    }
-}
-
 impl FullBacktraceCollector {
-    pub fn new() -> Self {
+    /// # Safety
+    /// This accesses the global [`CALLSTACKS`] variable and may not be called concurrently.
+    pub unsafe fn new() -> Self {
         unsafe { (*addr_of_mut!(CALLSTACKS)) = Some(ThreadLocal::new()) };
         Self {}
     }
 
     pub fn reset(&mut self) {
+        // # Safety
+        // This accesses the global [`CALLSTACKS`] variable.
+        // While it is racey, it might be fine if multiple clear the vecs concurrently.
+        // TODO: This should probably be rewritten in a safer way.
         unsafe {
```

Comment:
This is actually shady and could probably be rewritten in a safer way? Technically this function needs to be marked unsafe, but it propagates up a lot.. cc @rmalmain

Reply:
yes, [this is definitiely easy to break](https://godbolt.org/z/zM8Gexn1v)
we should just return a full vector (not a ref) at this point; it's cloned later on anyway in every caller of the `backtrace` function.